### PR TITLE
Fix messages intermittently disappearing when first opening the channel

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
https://github.com/GetStream/stream-chat-swift/issues/2423
https://github.com/GetStream/ios-issues-tracking/issues/22

### 🎯 Goal
Fixes an issue that messages would disappear when opening a channel for the first time. This is mostly reproducible by changing the `messageLimit = 1` in the `ChannelListQuery`. 

**Bonus:** It also fixes the first page not being loaded from the cache when using `messageLimit = 1`

### 📝 Summary
- Fixes messages intermittently disappearing when first opening the channel
- Fixes the first page not loaded from the cache when using a lower message limit in `ChanneListQuery`

### 🛠 Implementation
It basically reverts #2298, this PR was meant to fix quoted and pinned messages from incorrectly appearing on the first page of the message list, but it caused the 2 issues above.

In order to still fix the original issue with #2298, a fetchLimit and batchSize were added to the CoreData fetch request to make sure the messages are always fetched by batches of the page size. And so, the first page will only fetch 25 messages on the first page, filtering out the pinned or quoted messages that are in the DB. Since the next page will also be 25, and the messages are ordered by `createdAt`, the pinned and quoted messages will always be filtered out, until they actually appear on the page from the network.

### 🧪 Manual Testing Notes
Steps to reproduce the missing messages:
1. Change messageLimit of the Channel List Query to 1 (Right now this can only be done in code, but will soon add a configuration to the DemoApp)
2. Open a Channel
3. Scroll to top and load 1 or 2 more pages
4. Leave the app
5. Open the app
6. Open the same channel
7. Only one message appears
8. Scroll to top
9. One page is skipped, missing messages

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Database/DTOs/MessageDTO.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Database/DTOs/MessageDTO.swift:235]:
```suggestion
        // Some quoted or pinned messages might be in the local database, but should not be fetched
```

File: Sources/StreamChat/Database/DTOs/ChannelDTO.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Database/DTOs/ChannelDTO.swift:539]: Note to self: Use a guard here and explain why we need `isOlderThanCurrentOldestMessage`

File: Sources/StreamChat/Workers/ChannelUpdater.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Workers/ChannelUpdater.swift:46]: Note to self: only call `session.channel(cid: payload.channel.cid)` once

File: Sources/StreamChat/Controllers/MessageController/MessageController.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Controllers/MessageController/MessageController.swift:621]: Note to self: This needs to come from the MessageController Query or page configuration of some sort

**Review by nuno-vieira — COMMENTED** [review]
Still, some improvements are required before merging. As well as making the messageLimit configurable in the DemoApp

File: Sources/StreamChat/Workers/ChannelUpdater.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Workers/ChannelUpdater.swift:50]: We can actually reset the `oldestMessageAt` here and it will also fix the problem 🤔 Not requiring the fetchLimit and batchSize

File: CHANGELOG.md
Comment (martinmitrevski) [inline CHANGELOG.md:12]: Also, please make sure that we mention that only the first page is loaded in offline mode now.

File: Sources/StreamChat/Controllers/MessageController/MessageController.swift
Comment (martinmitrevski) [inline Sources/StreamChat/Controllers/MessageController/MessageController.swift:621]: Did you already change this (seems it's the same after you put the note)?

File: Sources/StreamChat/Database/DTOs/ChannelDTO.swift
Comment (martinmitrevski) [inline Sources/StreamChat/Database/DTOs/ChannelDTO.swift:533]: the comment here is a bit confusing, can you please rephrase it?

**Review by martinmitrevski — COMMENTED** [review]
LGTM! Would be good if @alteral does some in depth QA, since it's a tricky change.

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅ Just a small update to the changelog is needed (more details on Slack).

**Comment by alteral** [comment]
QAed, LGTM ✅

Only two concerns:
- `test_controller_reportsInitialValues_forDMChannel_ifChannelExistsLocally` is failing on CI
- There is a tiny bug when the last message is separated from the others after reconnection. Might be unrelated to this PR, but can't be checked on `develop` because of the original bug

File: Sources/StreamChat/Database/DTOs/ChannelDTO.swift
Comment (polqf) [inline Sources/StreamChat/Database/DTOs/ChannelDTO.swift:323]: Why are we not clearing this value when removing the channel?

**Review by polqf — APPROVED** [review]

File: Sources/StreamChat/Utils/LazyCachedMapCollection.swift
Comment (nuno-vieira) [inline Sources/StreamChat/Utils/LazyCachedMapCollection.swift:36]:
```suggestion
        // that these DTOs are accessed on their managed object contexts
```

**Pull request merged**
